File: 卡码/代码随想录/1/03-哈希表/Python/03-快乐数/LeetCode_0202.py
#202. 快乐数
from typing import *

class Solution:
	def getSum(self,n:int)->int:
		sum=0
		while n>0:
			sum+=(n%10)**2
			n//=10
		return sum
	def isHappy(self, n: int) -> bool:
		s=set()
		while n!=1 and n not in s:
			s.add(n)
			n=self.getSum(n)
		return n==1

File: 卡码/代码随想录/1/03-哈希表/Js/03-快乐数/LeetCode_0202.js
//202. 快乐数
const getSum=n=>{
  let sum=0;
  while(n>0){
    sum+=(n%10)*(n%10);
    n=Math.floor(n/10);
  }
  return sum;
}
const isHappy=n=>{
  let set=new Set();
  while(n!=1&&!set.has(n)){
    console.log(set);
    set.add(n);
    n=getSum(n);
  }
  return n==1;
}